Gopkg.toml

```diff
@@ -18,10 +18,34 @@
   name = "k8s.io/apiextensions-apiserver"
   version = "kubernetes-1.11.2"
 
+[[override]]
+  name = "k8s.io/apiserver"
+  version = "kubernetes-1.11.2"
+
 [[override]]
   name = "k8s.io/client-go"
   version = "kubernetes-1.11.2"
 
+[[override]]
+  name = "k8s.io/kubernetes"
+  version = "1.11.2"
+
+[[constraint]]
+  name = "k8s.io/helm"
+  version = "2.11.0"
+
+[[override]]
+  name = "github.com/russross/blackfriday"
```

Comment:
Why do we need to override this? Just curious about the reason behind this. Thanks!

Reply:
`github.com/russross/blackfriday` is depended on by `k8s.io/kubernetes`, which doesn't have dep files. I'm not totally familiar with how dep chooses dependencies, but in this case it chooses the latest tag (v2.0.1) which is incompatible with `k8s.io/kubernetes` usage of it. I'll update this to specify the latest v1 tag (1.5.2), which does work.

`github.com/docker/distribution` is also depended on by `k8s.io/kubernetes`. The latest stable tag in that repo is 2.6.2, which is what dep chooses without an override. That version is not compatible. There is a 2.7.0 pre-release tag that would work. I'm not sure if `2.7.0-rc.0` or `master` would be better.

`github.com/docker/docker` is also depended on by `k8s.io/kubernetes`. Without the override, dep chooses 1.6.0-rc5, which is incompatible with `k8s.io/kubernetes` usage of it. The main problem with more recent tagged versions is that they use `github.com/Sirupsen/logrus` and we use `github.com/sirupsen/logrus`, so dep refuses to pull a more recent version. The latest master switches to `github.com/sirupsen/logrus`, but for some reason, docker devs don't seem to be tagging that repo anymore.

I also noticed that I need to use "=1.11.2" instead of "1.11.2" for the kubernetes override to force it to pick 1.11.2 exactly, so I'll make that change as well.

Reply:
It's a shame we need to import `k8s.io/kubernetes` in the first place just to satisfy tillers need for the different clients it uses.

But that made me look at https://github.com/helm/helm/blob/2e55dbe1fdb5fdb96b75ff144a339489417b146b/glide.lock in case you find some inspiration there. 

All makes sense, thanks for the explanation! Maybe we could add that to the commit msg or a short comment in the Gopkg.toml? WDYT?